api/hasura/actions/_handlers/createUserWithToken.ts

```diff
@@ -0,0 +1,71 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import { getAddress } from '../../../../api-lib/gql/queries';
+import { UnprocessableError } from '../../../../api-lib/HttpError';
+import { verifyHasuraRequestMiddleware } from '../../../../api-lib/validate';
+import { CircleTokenType } from '../../../../src/common-lib/circleShareTokens';
+import {
+  createUserFromTokenInput,
+  HasuraUserSessionVariables,
+  composeHasuraActionRequestBodyWithSession,
+} from '../../../../src/lib/zod';
+
+import { createUserMutation } from './createUserMutation';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  const {
+    input: { payload: input },
+    session_variables: sessionVariables,
+  } = composeHasuraActionRequestBodyWithSession(
+    createUserFromTokenInput,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+
+  // get address from currrent user
+  const { hasuraProfileId } = sessionVariables;
+  console.log('getting addr');
+  const address = await getAddress(hasuraProfileId);
+
+  console.log('getting tokens');
+  // get the circleId from the token and make sure its magic
+  const { circle_share_tokens } = await adminClient.query({
+    circle_share_tokens: [
+      {
+        where: {
+          uuid: {
+            _eq: input.token,
```

Comment:
might want to validate if the circle is already soft deleted

Reply:
addressed